File: task7/task7.cpp
#include <iostream>
#include <string>

std::string rot13(const std::string& msg) {
    const int LEN_ALPHABET = 'z' - 'a' + 1;
    std::string result = "";

    for (char c : msg) {
        if (islower(c) != 0) {
            result.push_back('a' + (c - 'a' + 13) % LEN_ALPHABET);
        } else if (isupper(c) != 0) {
            result.push_back('A' + (c - 'A' + 13) % LEN_ALPHABET);
        } else {
            result.push_back(c);
        }
    }
    
    return result;
}

int main() {
    std::cout << rot13("test") << '\n';  // "grfg"
    std::cout << rot13("Test") << '\n';  // "Grfg"
    std::cout << rot13("AbCd") << '\n';  // "NoPq"
    std::cout << rot13("") << '\n';  // ""
    std::cout << rot13("4") << '\n';  // "4"
    std::cout << rot13("Z") << '\n';  // "M"
    std::cout << rot13("t89est") << '\n';  // "g89rfg"
    std::cout << rot13("Test6") << '\n';  // "Grfg6"

    return 0;
}

File: archive/task07/task7.py
def rot13(msg):
    LEN_ALPHA = ord('z') - ord('a') + 1
    result = ''

    for c in msg:
        if c.islower():
            result += chr(ord('a') + (ord(c) - ord('a') + 13) % LEN_ALPHA)
        elif c.isupper():
            result += chr(ord('A') + (ord(c) - ord('A') + 13) % LEN_ALPHA)
        else:
            result += c

    return result


print(rot13("test"))  # "grfg"
print(rot13("te665st"))  # "gr665fg"
print(rot13("4"))  # "4"
print(rot13("Test"))  # "Grfg"
